protozoo/trainer.py

```diff
@@ -1,33 +1,43 @@
 from ignite.engine import Events, Engine
 
-from protozoo.tiktorch_config_keys import ModelZooEntry
+from protozoo.config import ModelZooEntry
+from protozoo.shared_config import MapTo
 
 
-def get_trainer(model_zoo_entry: ModelZooEntry) -> Engine:
-    def training_step(trainer: Engine, batch) -> float:
+class Trainer(Engine):
+    def __init__(self, model_zoo_entry: ModelZooEntry):
+        model_zoo_entry = model_zoo_entry.get_mapped(MapTo.PYTORCH)
+        self.model_config = model_zoo_entry.model_config
+        self.optimizer_config = model_zoo_entry.optimizer_config
+        self.loss_config = model_zoo_entry.loss_config
+
+        if self.model_config.pretrained_source is not None:
+            raise NotImplementedError("model_zoo_entry.model_config.pretrained_source")
+
+        super().__init__(self.training_step)
+
+        self.add_event_handler(Events.STARTED, self.setup)
+        for callback in model_zoo_entry.trainer_callbacks:
```

Comment:
I am not against making config in such a way so it would be possible to do this, but I am against actually implementing it